### ISSUE TITLE
Showing images inline in messaging window

### ISSUE THREAD

**Issue by zoltan-dulac** [dylex/slack-libpurple#15]
I know this is on the list of thing to do, but I'd like to up-vote it via here.  Note that it is not horrible if it's not available right away, since the URLs are now showing up when the image isn't available, which is the way the Slack's XMPP bridge handles it.  However, it would be great if it were implemented.

One of things I have noticed with Pidgin plugins that do inline images (i.e. https://bitbucket.org/EionRobb/purple-hangouts) is that sometimes pidgin will crash (rarely, but it happens).  Maybe it would be nice to have a way to turn on or off inline images?

**Comment by fredcooke** [comment]
Agreed re on/off control as bandwidth may be an issue. Similar, what I would prefer is for it to grab them for me into some predictable place (folder in account setup perhaps) where I can view them locally and keep a copy without manual download. This is particularly important with free accounts as I do this religiously as those files/uploads go away with time as newer content replaces them within the free account limits. I can make a new ticket for this if @dylex or co wants me to. Maybe I could even help, but not right now.

**Comment by WillForan** [comment]
Any ideas for how to start?
Maybe in [`slack_attachment_to_html`](https://github.com/dylex/slack-libpurple/blob/master/slack-message.c#L209) (references [api]( https://api.slack.com/docs/message-attachments)) with cues taken from [`purple_image_new_from_data`](https://github.com/EionRobb/purple-hangouts/blob/master/purple2compat/image.h#L8) (redefined [`purple_imgstore_add`](https://github.com/Tasssadar/libpurple/blob/master/imgstore.c#L39)) in [`purple-hangouts/hangouts_events.c`](https://github.com/EionRobb/purple-hangouts/blob/master/hangouts_events.c#L540)?
